File: Ejercicios/Ejercicio2/2.java
import java.util.*;

class Evento 
{
    String nombre;
    String fecha;
    List<String> asistentes;

    public Evento(String nom, String fec) 
  {
        nombre = nom;
        fecha = fec;
        asistentes = new ArrayList<>();
    }

    public void agregarAsistente(String asistente)
  {
        asistentes.add(asistente);
        Collections.sort(asistentes);
    }

    public void mostrarAsistentes() 
  {
        for (String asistente : asistentes)
          {
            System.out.println(asistente);
        }
    }
}

class Organizador
  {
    List<Evento> eventos;

    public Organizador()
    {
        eventos = new ArrayList<>();
    }

    public void agregarEvento(String nombre, String fecha) 
    {
        boolean existe = eventos.stream().anyMatch(evento -> evento.nombre.equals(nombre));
        if (existe)
        {
            System.out.println("Evento con el nombre " + nombre + " ya existe.");
            return;
        }
        eventos.add(new Evento(nombre, fecha));
        System.out.println("Evento agregado exitosamente.");
    }

    public void agregarAsistente(String nombreEvento, String asistente)
    {
        Optional<Evento> evento = eventos.stream().filter(e -> e.nombre.equals(nombreEvento)).findFirst();
        if (evento.isPresent()) 
        {
            evento.get().agregarAsistente(asistente);
            System.out.println("Asistente agregado exitosamente.");
        } 
        else 
        {
            System.out.println("Evento no encontrado.");
        }
    }

    public void mostrarAsistentes(String nombreEvento)
    {
        Optional<Evento> evento = eventos.stream().filter(e -> e.nombre.equals(nombreEvento)).findFirst();
        if (evento.isPresent()) 
        {
            evento.get().mostrarAsistentes();
        } 
        else
        {
            System.out.println("Evento no encontrado.");
        }
    }

    public void buscarEventosPorFecha(String fecha)
    {
        boolean encontrado = false;
        for (Evento evento : eventos) 
        {
            if (evento.fecha.equals(fecha)) 
            {
                System.out.println("Evento: " + evento.nombre);
                encontrado = true;
            }
        }
        if (!encontrado) 
        {
            System.out.println("No se encontraron eventos en la fecha " + fecha);
        }
    }
}

public class Main {
    public static void mostrarMenu()
  {
        System.out.println("Menu de Gestion de Eventos");
        System.out.println("1. Agregar evento");
        System.out.println("2. Agregar asistente a evento");
        System.out.println("3. Mostrar asistentes de un evento");
        System.out.println("4. Buscar eventos por fecha");
        System.out.println("5. Salir");
    }

    public static void main(String[] args)
  {
        Organizador organizador = new Organizador();
        Scanner scanner = new Scanner(System.in);
        int opcion;

        do {
            mostrarMenu();
            System.out.print("Seleccione una opcion: ");
            opcion = scanner.nextInt();

            scanner.nextLine(); 
            String nombre, fecha, asistente;

            switch (opcion) {
                case 1:
                    System.out.print("Ingrese nombre del evento: ");
                    nombre = scanner.nextLine();
                    System.out.print("Ingrese fecha del evento: ");
                    fecha = scanner.nextLine();
                    organizador.agregarEvento(nombre, fecha);
                    break;
                case 2:
                    System.out.print("Ingrese nombre del evento: ");
                    nombre = scanner.nextLine();
                    System.out.print("Ingrese nombre del asistente: ");
                    asistente = scanner.nextLine();
                    organizador.agregarAsistente(nombre, asistente);
                    break;
                case 3:
                    System.out.print("Ingrese nombre del evento: ");
                    nombre = scanner.nextLine();
                    organizador.mostrarAsistentes(nombre);
                    break;
                case 4:
                    System.out.print("Ingrese fecha del evento (YYYY-MM-DD): ");
                    fecha = scanner.nextLine();
                    organizador.buscarEventosPorFecha(fecha);
                    break;
                case 5:
                    System.out.println("Saliendo...");
                    break;
                default:
                    System.out.println("Opcion no valida. Intente nuevamente.");
            }
        } while (opcion != 5);

        scanner.close();
    }
}

File: Ejercicios/Ejercicio2/2.py
class Evento:
    def __init__(self, nom, fec):
        self.nombre = nom
        self.fecha = fec
        self.asistentes = []

    def agregarAsistente(self, asistente):
        self.asistentes.append(asistente)
        self.asistentes.sort()

    def mostrarAsistentes(self):
        for asistente in self.asistentes:
            print(asistente)

class Organizador:
    def __init__(self):
        self.eventos = []

    def agregarEvento(self, nombre, fecha):
        if any(evento.nombre == nombre for evento in self.eventos):
            print(f"Evento con el nombre {nombre} ya existe.")
            return
        self.eventos.append(Evento(nombre, fecha))
        print("Evento agregado exitosamente.")

    def agregarAsistente(self, nombreEvento, asistente):
        for evento in self.eventos:
            if evento.nombre == nombreEvento:
                evento.agregarAsistente(asistente)
                print("Asistente agregado exitosamente.")
                return
        print("Evento no encontrado.")

    def mostrarAsistentes(self, nombreEvento):
        for evento in self.eventos:
            if evento.nombre == nombreEvento:
                evento.mostrarAsistentes()
                return
        print("Evento no encontrado.")

    def buscarEventosPorFecha(self, fecha):
        encontrados = False
        for evento in self.eventos:
            if evento.fecha == fecha:
                print(f"Evento: {evento.nombre}")
                encontrados = True
        if not encontrados:
            print(f"No se encontraron eventos en la fecha {fecha}")

def mostrarMenu():
    print("Menu de Gestion de Eventos")
    print("1. Agregar evento")
    print("2. Agregar asistente a evento")
    print("3. Mostrar asistentes de un evento")
    print("4. Buscar eventos por fecha")
    print("5. Salir")

    organizador = Organizador()
    opcion = 0

    while opcion != 5:
        mostrarMenu()
        opcion = int(input("Seleccione una opcion: "))

        if opcion == 1:
            nombre = input("Ingrese nombre del evento: ")
            fecha = input("Ingrese fecha del evento: ")
            organizador.agregarEvento(nombre, fecha)
        elif opcion == 2:
            nombre = input("Ingrese nombre del evento: ")
            asistente = input("Ingrese nombre del asistente: ")
            organizador.agregarAsistente(nombre, asistente)
        elif opcion == 3:
            nombre = input("Ingrese nombre del evento: ")
            organizador.mostrarAsistentes(nombre)
        elif opcion == 4:
            fecha = input("Ingrese fecha del evento (YYYY-MM-DD): ")
            organizador.buscarEventosPorFecha(fecha)
        elif opcion == 5:
            print("Saliendo...")
        else:
            print("Opcion no valida. Intente nuevamente.")
